www_src/lib/api.js

```diff
@@ -200,4 +233,39 @@ api.signUp = function (options, callback) {
   });
 };
 
+/**
+ * [getElement Retrieves an element from the db]
+ * @param  {Object}   options  Should include ids for: user, project, page, and element
+ * @param  {Function} callback Called with two params, err and data,
+ *                             where err.message is a descriptive error message if an error occured,
+ *                             or data is an object representing the element.
+ */
+api.getElement = function (options, callback) {
+  api.validateOptions('getElement', ['user', 'project', 'page', 'element'], options);
```

Comment:
this array feels a bit magical. Can we refactor these to a mapping that can be maintained without needing to go into functions? Like
```
var requirements = {
  'getElement':  ['user', 'project', 'page', 'element'],
  'updateElement': ['user', 'project', 'page', 'element', 'json'],
  ...
};
```
and then refer to that here, like
```
  api.validateOptions('getElement', requirements['getElement'], options);
```
or just
```
  api.validateOptions('getElement', options);
```

with the validateOptions function pulling the necessary requirements?